src/main/java/decodes/db/HasIterator.java

```diff
@@ -0,0 +1,8 @@
+package decodes.db;
```

Comment:
This isn't the right package location. Also I think I need a few more interfaces to help simplify the Calcite code a little.

Once more discussion has happened we'll figure out a better location.